### ISSUE TITLE
Add a get_statistics() method

### ISSUE THREAD

**Issue by anirudt** [scikit-learn-contrib/metric-learn#30]
Hi,

Is there any roadmap for adding a get_statistics() function, like the one that Shogun offers, which helps us to get the statistics of a certain model? It could be useful like [here](http://nbviewer.jupyter.org/gist/iglesias/6576096), where the nature of the objective function could be a useful insight to assigning a certain value of `maxiters`.

**Comment by perimosocordiae** [comment]
I haven't considered keeping track of per-iteration objective function values, though I see why it might be useful in some cases.

My main concern is that this might negatively impact time/memory performance for cases where the user doesn't need the training statistics. I'd like there to be a parameter that controls whether these sorts of debugging/tuning statistics are collected.

**Comment by anirudt** [comment]
I agree. Maybe we can have a time/memory complexity profiling and decide likewise? I could try my hand on this.

**Comment by bellet** [comment]
This has been inactive for a very long time so I will close it for now
